design/ordinal-crm.Rmd

```diff
@@ -0,0 +1,453 @@
+Hall\
+<https://www.utstat.toronto.edu/~brunner/oldclass/2201s11/readings/glmbook.pdf>
+
+[12] Van Meter EM, Garrett-Mayer E, Bandyopadhyay D. Dose-finding clinical trial design for ordinal
+toxicity grades using the continuation ratio model: an extension of the continual reassessment
+method. Clin Trials. 2012 Jun;9(3):303-13. doi: 10.1177/1740774512443593. Epub 2012 Apr 30.\
+<https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5531273/>
+
+[13] Wikipedia, <https://en.wikipedia.org/wiki/Truncated_normal_distribution>.
+
+[14] Greene, William H. (2003). Econometric Analysis (5th ed.). Prentice Hall.
+
+[15] Horrace, William C., "Some Results on the Multivariate Truncated Normal Distribution" (2005).
+Economics - Faculty Scholarship. 19.
+
+[16] Jack Cartinhour (1990) One-dimensional marginal density functions of a truncated multivariate
+normal density function, Communications in Statistics - Theory and Methods, 19:1, 197-203, DOI:
+10.1080/03610929008830197
+
+# Prototype code
+
+## Fictitious observed data
+
+```{r data}
+dose_grid <- c(5, 15, 45, 70, 100, 220, 300, 600, 1000, 1800, 4000, 10000, 16000)
+
+x <- c(
+  rep(5, 1),
+  rep(15, 4),
+  rep(45, 5),
+  rep(70, 5),
+  rep(100, 5),
+  rep(220, 8),
+  rep(300, 6),
+  rep(600, 15),
+  rep(1000, 8),
+  rep(1800, 9),
+  rep(4000, 10),
+  rep(10000, 14),
+  rep(16000, 2)
+)
+
+y <- c(
+  rep(0, 1),
+  rep(0, 4),
+  rep(0, 5),
+  rep(0, 5),
+  rep(0, 5),
+  rep(0, 7), 1,
+  rep(0, 5), 1,
+  rep(0, 10), 1, 1, 1, 1, 1,
+  rep(0, 5), 1, 1, 1,
+  rep(0, 8), 1,
+  rep(0, 8), 1, 1,
+  rep(0, 10), 1, 1, 1, 1,
+  rep(1, 1),
+  rep(2, 1)
+)
+```
+
+## Data-class
+
+We need to adopt a new class for data as the existing one `Data` does not allow for categorical (\>2
+categories) toxicity. We propose to create new data class, named e.g. `DataCatTox`, which inherits
+from existing `GeneralData` class and is a parent of existing `Data` class. The new `DataCatTox`
+class:
+
+-   gets its slots from current `Data` and
+-   adds one new slot `yCategories`. It is a named integer vector that keeps all the possible values
+    for `y`. Default to `c("No DLT" = 0L, "DLT" = 1L)`.
+
+Eventually, instead of having `yCategories` as a named vector, we can introduce a new slot, e.g.
+`yLabels`. Below is the prototype code.
+
+```{r prototype_data, eval=FALSE}
+.DataCatTox <- setClass(
+  Class = "DataCatTox",
+  contains = "GeneralData",
+  slots = c(
+    x = "numeric",
+    y = "integer",
+    doseGrid = "numeric",
+    nGrid = "integer",
+    xLevel = "integer",
+    yCategories = "integer",
+    placebo = "logical"
+  ),
+  prototype = prototype(
+    x = numeric(),
+    y = integer(),
+    doseGrid = numeric(),
+    nGrid = 0L,
+    xLevel = integer(),
+    yCategories = c("No DLT" = 0L, "DLT" = 1L),
+    placebo = FALSE
+  ),
+  validity = v_data_cat_tox
+)
+
+.Data <- setClass(
+  Class = "Data",
+  contains = "DataCatTox",
```

Comment:
hm not sure if this will be easy... I am a bit afraid of downstream implications of this. could be worth a try but let's be ready to fail fast if it turns out to be too complicated. I think just having `DataOrdinal` be parallel to `Data` class would be safer.